### PR TITLE
[backend] Fix latency scenarioId

### LINKED ISSUE #1582: Improve latency on endpoint: admin/scenarios/:id
## Use case

Latency spotted: 
- /api/scenarios/322e7f9d-e002-4eac-a402-a01446f4b1f4/exercises

### PR DESCRIPTION
**Pull request by savacano28**
### Proposed changes

*
*

### Related issues

* Closes #1582 
*

### Checklist

- [x] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: openbas-model/src/main/java/io/openbas/database/repository/EndpointRepository.java [hunk 1]
```diff
@@ -2,8 +2,10 @@
 
 import io.openbas.database.model.Endpoint;
 import io.openbas.database.model.Scenario;
+import io.openbas.database.raw.RawAsset;
 import jakarta.validation.constraints.NotBlank;
 import jakarta.validation.constraints.NotNull;
+import org.springframework.data.jpa.domain.Specification;
```
Comment (damgouj) [inline openbas-model/src/main/java/io/openbas/database/repository/EndpointRepository.java:8]: Unused imports

**Review by damgouj — COMMENTED** [review]
Great job ! I tried to do my best to check some scenarios and all the code updated.

**Review by damgouj — APPROVED** [review]

File: openbas-model/src/main/java/io/openbas/database/repository/ExerciseRepository.java [hunk 2]
```diff
@@ -20,170 +20,244 @@ public interface ExerciseRepository extends CrudRepository<Exercise, String>,
...
+          + "ex.exercise_subtitle, "
+          + " array_agg(distinct ie.inject_id) FILTER ( WHERE ie.inject_id IS NOT NULL ) as inject_ids, "
+          + " array_agg(distinct et.tag_id) FILTER ( WHERE et.tag_id IS NOT NULL ) as exercise_tags "
+          + "FROM exercises ex "
+          + "LEFT JOIN scenarios_exercises s ON s.exercise_id = ex.exercise_id "
+          + "LEFT JOIN exercises_tags et ON et.exercise_id = ex.exercise_id "
+          + "LEFT JOIN injects_expectations ie ON ex.exercise_id = ie.exercise_id "
+          + "WHERE s.scenario_id IN (:scenarioIds) "
+          + "GROUP BY ex.exercise_id ;", nativeQuery = true)
+  List<RawExercise> rawAllByScenarioId(@Param("scenarioIds") List<String> scenarioIds);
```
Comment (RomuDeuxfois) [inline openbas-model/src/main/java/io/openbas/database/repository/ExerciseRepository.java:261]: Same here, the query is not aligned with properties on RawExercise.
It's a big one

File: openbas-model/src/main/java/io/openbas/database/repository/ExerciseRepository.java [hunk 3]
```diff
@@ -20,170 +21,251 @@ public interface ExerciseRepository extends CrudRepository<Exercise, String>,
...
+          + "ex.exercise_category, "
+          + "ex.exercise_subtitle, "
+          + " array_agg(distinct ie.inject_id) FILTER ( WHERE ie.inject_id IS NOT NULL ) as inject_ids, "
+          + " array_agg(distinct et.tag_id) FILTER ( WHERE et.tag_id IS NOT NULL ) as exercise_tags "
+          + "FROM exercises ex "
+          + "LEFT JOIN exercises_tags et ON et.exercise_id = ex.exercise_id "
+          + "LEFT JOIN injects_expectations ie ON ex.exercise_id = ie.exercise_id "
+          + "WHERE ex.exercise_id = :id "
+          + "GROUP BY ex.exercise_id ;", nativeQuery = true)
+  RawExercise rawById(@Param("id") String id);
```
Comment (RomuDeuxfois) [inline openbas-model/src/main/java/io/openbas/database/repository/ExerciseRepository.java:250]: You can remove this one, let's talk about it.

File: openbas-model/src/main/java/io/openbas/database/raw/RawUser.java [hunk 4]
```diff
@@ -11,14 +10,17 @@ default String getUser_gravatar() {
...
   String getUser_lastname();
+
   String getUser_email();
+
   String getUser_phone();
-  Instant getUser_created_at();
+
   String getUser_organization();
-  List<String> getUser_groups();
-  List<String> getUser_teams();
```
Comment (RomuDeuxfois) [inline openbas-model/src/main/java/io/openbas/database/raw/RawUser.java:21]: This is populate by the   List<RawUser> rawAll(); query

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**